src/Evaluators/MOI_wrapper.jl

```diff
@@ -0,0 +1,75 @@
+# MOI wrapper
+
+mutable struct ExaEvaluator <: MOI.AbstractNLPEvaluator
```

Comment:
That sounds now totally picky and maybe unnecessary, but you think we shouldn't call it `ExaPFEvaluator` or maybe even just `Evaluator` like in JuMP?

Reply:
No, this is particularly relevant. Indeed, I should be more careful with naming. I have renamed it `MOIEvaluator`, to emphasis the fact that we rely on MOI there.